packages/moonstone/internal/DateTimeDecorator/DateTimeDecorator.js

```diff
@@ -76,42 +95,14 @@ const DateTimeDecorator = hoc((config, Wrapped) => {
 			}
 		}
 
-		UNSAFE_componentWillReceiveProps (nextProps) {
-			const newValue = this.toTime(nextProps.value);
-
-			if (nextProps.open && !this.props.open) {
-				const value = newValue || Date.now();
-
-				// if we're opening, store the current value as the initial value for cancellation
-				this.setState({
-					initialValue: newValue,
-					pickerValue: null,
-					value
-				});
-
-				// if no value was provided, we need to emit the onChange event for the generated value
-				if (!newValue) {
-					this.emitChange(this.toIDate(value));
-				}
-			} else {
-				this.setState({
-					value: newValue
-				});
-			}
-		}
-
-		UNSAFE_componentWillUpdate () {
-			// check for a new locale when updating
-			this.initI18n();
-		}
-
-		initI18n () {
-			const locale = ilib.getLocale();
-
-			if (i18n && this.locale !== locale && typeof window === 'object') {
-				this.locale = locale;
-				this.i18nContext = i18n();
+		static getDerivedStateFromProps (props, state) {
+			const propValue = toTime(props.value);
+			if (state.value !== propValue) {
+				return {
+					value: propValue
```

Comment:
Is the convention `prevValue`?

Reply:
The intention here is not to cache, but only to convert the current prop value.